### PR TITLE
feat(binary-schemas): add new 'binary-schemas' rule

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit introduces the new spectral-style
'binary-schemas' rule, which replaces the legacy
'json_or_param_binary_string' rule.
While testing this new rule, I discovered a bug
in the 'response-body-object' rule where it enforces
"request body must be an object" for non-JSON request
bodies.  This was fixed by updating the rule's
"given" field with a more accurate json path expression.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

#### Checklist for removing an old validation rule:
- [x] Removed old rule implementation (packages/validator/src/plugins/validation/*.js)
- [x] Removed or adjusted testcases (packages/validator/test)
- [x] Updated default configuration to deprecate old rule (packages/validator/src/.defaultsForValidator.js)
- [x] Removed docs of old rule (docs/ibm-legacy-rules.md)

### CODE REVIEW

File: packages/ruleset/src/rules/request-body-object.js [hunk 1]
```diff
@@ -2,7 +2,8 @@ const { enumeration } = require('@stoplight/spectral-functions');
 
 module.exports = {
   description: 'All request bodies MUST be structured as an object',
-  given: '$.paths[*][*].requestBody.content[*].schema',
+  given:
+    '$.paths[*][*].requestBody.content.[?(@property ~= "^application\\\\/json(;.*)*$")].schema',
```
Comment (padamstx) [inline packages/ruleset/src/rules/request-body-object.js:6]: This changes the rule so it only enforces the rule on JSON content.

**Review by pyrooka — APPROVED** [review]
LGTM!

**Review by dpopp07 — APPROVED** [review]
Looks good! Left one comment about where we might want to hold off to take advantage of Dan's utility functions but all looks good otherwise.

Also, that's a lot of legacy files removed! Nice!

File: packages/ruleset/src/functions/binary-schemas.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        path
+      }
+    ];
+  }
+
+  return [];
+}
+
+// Returns true if "schema" is a binary schema (raw octet stream, file).
+function isBinarySchema(schema) {
```
[the remark this answers is not in the document]
Comment (padamstx) [reply packages/ruleset/src/functions/binary-schemas.js]: done

**Pull request merged**